### ISSUE TITLE
Need a way to dnd to top/bottom of page

### ISSUE THREAD

**Issue by doronrosenberg** [maqetta/maqetta#2180]
When we have full screen widgets (like ScrollableView), it seems impossible to drop something at the top or bottom of the page.

@jhpedemonte @JonFerraiolo 

Perhaps when we are in DND, we overlay on top of the toolbar 2 drop targets - one for dropping at the top, one for the bottom of the page?

Another option is that fullscreen widgets have drop areas at the top/bottom which allow to drop before/after.

**Comment by jhpedemonte** [comment]
I think we need more scenarios here.  The one you mention (dojox.mobile Views) is somewhat of a special case -- the Views are (for the most part) meant to take the full screen and then just act as containers for subsequent widgets.  This is how they'll be used 99% of the time.

There is the rare case where a user will want to put a widget outside of the view -- for example, Header and TabBar widgets can be siblings to ScrollableView, where they stay constant when switching between views.  In this case, we can just tell the user to make use of the Outline view.

At one point, I'd wanted to make it possible to drag/drop/add widgets from the Widgets palette directly on to the Outline view, to give finer control over situations such as this one.  But currently, the Outline view is hidden when the Widgets palette is visible.  Perhaps we can change the UI around to allow drag/drop from Widgets to Outline.  Or, if a user hovers over the Outline tab, it becomes visible, allowing drag/drop/add.

**Comment by JonFerraiolo** [comment]
> I think we need more scenarios here.

@doronrosenberg - what is the exact scenario (or scenarios) where the current UI is falling short? What widget in particular did you want to drop and what exactly did you want to happen? For example, were you trying to drop a Heading or TabBar and place it just before or just after a ScrollableView? For example, were you trying to create the following:

```
BODY
   HEADING
   SCROLLABLEVIEW
```

**Comment by doronrosenberg** [comment]
@JonFerraiolo correct, I wanted to place a widget before/after a widget that takes over the entire screen.  Say you want 3 ScrollableViews and a Tabbar underneath - adding the tabbar after adding all 3 views requires going into the outline view and manually moving it.

**Comment by JonFerraiolo** [comment]
OK, thanks. I would say the most important thing is providing a good UI experience for adding header and footer elements. My best idea so far is to add some special-case logic to Maqetta's drag/drop logic (from widget palette to canvas) for particular widgets (probably Heading and TabBar: with 'fixed') such that when dragging near top of bottom, we give visual feedback to user that drop will go into top or bottom, and then upon dropping, we execute that. I haven't figured out the details, but that's the sketch I have in mind.

**Comment by jhpedemonte** [comment]
> ... requires going into the outline view and manually moving it.

Another reason to allow drag/drop into the Outline view itself.  However, not only isn't that implemented, it's just not feasible with our current layout, where the Widgets Palette and the Outline view overlay each other -- only one can be visible at a time.

**Comment by jhpedemonte** [comment]
-> M8

**Comment by JonFerraiolo** [comment]
-> M8 Prio-Med
